### PR TITLE
add Moco examples

### PR DESCRIPTION
**Pull request by carmichaelong**
@antoinefalisse @suhlrich Not sure how you all have been handling commits to the repo, so I'm pinging you both this time. Some main things to double check:
1. Is this the folder structure you all were thinking of?
2. What else might be useful in the README?

Of course, check out as much of the code that you'd like before merging. I'll also send out to some Moco users of varied experience level after merging too.

### CODE REVIEW

File: Moco/exampleSquat/exampleSquat.m [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+% Solve the problem
+% =================
+% Add a term to the solver that minimizes the implicit auxiliary
+% derivatives that are used with the implicit mode of muscles.
+solver = MocoCasADiSolver.safeDownCast(study.updSolver());
+solver.set_minimize_implicit_auxiliary_derivatives(true);
+solver.set_implicit_auxiliary_derivatives_weight(1e-6);
+
+solution = study.solve();
```
Comment (suhlrich) [inline Moco/exampleSquat/exampleSquat.m:255]: It'd be nice to add how long the example took to run on your computer

**Comment by carmichaelong** [comment]
@suhlrich thanks for the feedback. Added info in the comments at the top of the file.

File: Moco/exampleSquat/exampleSquat.m [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+% -----------
+% This goal enforces periodicity (i.e., enforce that certain values are
+% the same at the beginning of the simulation as the end of the
+% simulation).
+periodicityGoal = MocoPeriodicityGoal('periodicityGoal');
+problem.addGoal(periodicityGoal);
+model = modelProcessor.process();
+model.initSystem();
+
+% Periodic coordinate values (except for pelvis_tx) and speeds.
```
Comment (suhlrich) [inline Moco/exampleSquat/exampleSquat.m:153]: not obvious why you'd exclude tx and not tz?

File: Moco/exampleWalking/exampleWalking.m [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+% Set the weights for the terms in the objective function. The values below were
+% obtained by trial and error.
+controlEffortWeight = 0.1;
+stateTrackingWeight = 1;
+
+% Set other parameters, including the initial and final times of the
+% simulation, the mesh interval, the relative size of the state bounds, and
+% the optimal force for the lumbar and upper limb coordinate actuators.
+initialTime = 0.07;
```
Comment (suhlrich) [inline Moco/exampleWalking/exampleWalking.m:63]: maybe a little messy, but could be good to comment here and in the squats one how we select these times for periodicity constraints

File: Moco/exampleWalking/exampleWalking.m [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+problem = study.updProblem();
+
+
+% Goals
+% =====
+
+% Tracking
+% --------
+% Set different tracking weights for states (weights for states not 
+% explicitly set here have a default value of 1.0)
```
Comment (suhlrich) [inline Moco/exampleWalking/exampleWalking.m:110]: said these were determined by trial and error in the squat example. would be good to have here too.

**Review by suhlrich — APPROVED** [review]
Fantastic! Thanks for the awesome examples. Some very small documentation comments. Feel free to merge once done.

File: Moco/exampleSquat/exampleSquat.m (hunk 2 above)
Comment (carmichaelong) [reply Moco/exampleSquat/exampleSquat.m]: updated with explanation. although this works as is, could be good later to check the effect of also adding pelvis_tx periodicity and simplify the example if it doesn't have to be left out

Comment (suhlrich) [reply Moco/exampleSquat/exampleSquat.m]: helpful explanation. thanks. I get it now, and the choice makes sense.

**Pull request merged**